pkgs/servers/whisparr/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+
+, curl
+, dotnet-runtime
+, fetchurl
+, icu
+, libmediainfo
+, makeWrapper
+, mono
+, openssl
+, sqlite
+, zlib
+
+, nixosTests
+}:
+
+let
+  os = if stdenv.isDarwin then "osx" else "linux";
+  system = stdenv.hostPlatform.system;
+  arch = {
+    aarch64-linux = "arm64";
+    x86_64-darwin = "x64";
+    x86_64-linux = "x64";
+  }."${system}" or (throw "Unsupported system: ${system}");
+  hash = {
+    arm64-linux-hash = "sha256-cpKjYRQusNGZep2MdORCuEH9rCexqTtXwnsEqOz5C+E=";
+    x64-linux-hash = "sha256-RgJcTiO0JCjnYTO/kWfMjxnX3zwv48U6R/ARrjbJorU=";
+    x64-osx-hash = "sha256-ZvqfbFILplb01dxAkXsoBDhMUiBkP7VLClu5oQFbVxc=";
+  }."${arch}-${os}-hash";
+in
+stdenv.mkDerivation rec {
+  pname = "whisparr";
+  version = "2.0.0.287";
+
+  src = fetchurl {
+    name = "${pname}-${arch}-${os}-${version}.tar.gz";
+    url = "https://whisparr.servarr.com/v1/update/nightly/updatefile?runtime=netcore&version=${version}&arch=${arch}&os=${os}";
+    inherit hash;
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  runtimeLibs = lib.makeLibraryPath [
+    curl
+    icu
+    libmediainfo
+    mono
+    openssl
+    sqlite
+    zlib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    rm -rf "Whisparr.Update"
+
+    mkdir -p $out/{bin,share/${pname}-${version}}
+    cp -r * $out/share/${pname}-${version}/
+
+    makeWrapper "${dotnet-runtime}/bin/dotnet" $out/bin/Whisparr \
+      --add-flags "$out/share/${pname}-${version}/Whisparr.dll" \
+      --prefix LD_LIBRARY_PATH : ${runtimeLibs}
+
+    runHook postInstall
+  '';
+
+  passthru = {
+    updateScript = ./update.sh;
+    tests.smoke-test = nixosTests.whisparr;
+  };
+
+  meta = {
+    description = "Adult movie collection manager for Usenet and BitTorrent users";
+    homepage = "https://wiki.servarr.com/en/whisparr";
+    changelog = "https://whisparr.servarr.com/v1/update/nightly/changes";
+    license = lib.licenses.gpl3Only;
+    platforms = [ "aarch64-linux" "x86_64-darwin" "x86_64-linux" ];
```

Comment:
The [installation instructions](https://wiki.servarr.com/whisparr/installation/macos) mention an aarch64-darwin build. Is there a reason it is omitted from the package?